### ISSUE TITLE
Access violation in auto generated x:Bind code

### ISSUE THREAD

**Issue by sylveon** [microsoft/microsoft-ui-xaml#2572]
**Describe the bug**
The auto generated `x:Bind` code triggers an access violation.

Additionally, both `TemplateBinding` and `Binding` seems to fail here... `TemplateBinding` says "Setter does not have a value" (despite that being entirely false) and `Binding` says "The parameter is incorrect"

**Steps to reproduce the bug**
1. Clone and run https://github.com/sylveon/xaml-bug-repro
2. Observe access violation

**Expected behavior**
No access violation, the app runs successfully and shows the button.

**Screenshots**
![image](https://user-images.githubusercontent.com/6440374/83293195-6b555e80-a1b9-11ea-9dc2-dbc0b913c3f4.png)

**Version Info**

NuGet package version: 
[Microsoft.Windows.CppWinRT 2.0.200514.2]

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (19041)              | Yes   |
| November 2019 Update (18363)  |    |
| May 2019 Update (18362)            |    |
| October 2018 Update (17763)        |    |
| April 2018 Update (17134)          |    |
| Fall Creators Update (16299)       |    |
| Creators Update (15063)            |    |

| Device form factor | Saw the problem? |
| :-------------------- | :------------------- |
| Desktop                 | Yes   |
| Mobile                   |    |
| Xbox                      |    |
| Surface Hub          |     |
| IoT                         |     |

**Additional context**
I found this bug when writing XAML for an XAML Islands application, but it reproduces in UWP.

**Comment by sylveon** [comment]
I stepped through the code yesterday, and it seems the issue is this:
![](https://cdn.discordapp.com/attachments/580484470877061120/715810908776955964/unknown.png)

- `Connect` with id 6 acquires a reference to the `ColorAnimation` that has its property bound to a value
- `Connect` with id 2 updates all values, assuming all other connections where done prior and all references are valid

For some reason, `Connect` with id 2 is called but `Connect` with id 6 never got called prior to that, resulting in the framework trying to access a null pointer

**Comment by sylveon** [comment]
Some more updates:
- This happens in C# as well, both x:Bind and TemplateBinding
- Binding decided to work in C# and C++ now because ???

**Comment by sylveon** [comment]
Even more updates:
- Binding was working because I forgot I had changed the `Setter` for the `Normal` visual state...
- If I revert that change in C# and C++ UWP projects it still works, however my XAML Islands project crashes when doing that

**Comment by alwu-msft** [comment]
Yeah, they're deferred, but I would expect Connect() to be called when the transition is undeferred. Although I wonder if the undeferred ColorAnimation is then being immediately thrown away, rather than remaining undeferred... I'm not familiar with what VSM is doing under the covers so I don't know the answer off the top of my head.

**Comment by roxk** [comment]
Was about to report the same bug then found this issue. This issue is affecting `VisualState.StateTrigger` as well. In fact, I bet it's affecting anything that is bindable. 

A minimal repro for state trigger:
```
<Setter Property="Template">
            <Setter.Value>
                <ControlTemplate TargetType="local:CustomControl">
                    <Border
                        x:Name="border"
                        Background="{TemplateBinding Background}"
                        BorderBrush="{TemplateBinding BorderBrush}"
                        BorderThickness="{TemplateBinding BorderThickness}">
                        <VisualStateManager.VisualStateGroups>
                            <VisualStateGroup>
                                <VisualState x:Name="Normal"/>
                                <VisualState x:Name="PointerOver">
                                    <VisualState.Setters>
                                        <Setter Target="border.Background" Value="Red"/>
                                    </VisualState.Setters>
                                    <VisualState.StateTriggers>
                                        <StateTrigger IsActive="{x:Bind IsShown, Mode=OneWay}"/> // Crashes with null pointer deference when binding this state trigger
                                    </VisualState.StateTriggers>
                                </VisualState>
                            </VisualStateGroup>
                        </VisualStateManager.VisualStateGroups>
                        <TextBlock Text="Hello World"/>
                    </Border>
                </ControlTemplate>
            </Setter.Value>
        </Setter>
```
Did the same debugging sequence and could confirm the reference for the state trigger is never set in `Connect` prior to updating its value.

**Comment by sylveon** [comment]
This 4 years old bug is, hilariously, still an issue in WinUI 3

**Comment by sylveon** [comment]
However, the tags are incorrect. The bug was originally encountered in system XAML, with no WinUI at all.
